src/Cabal2nix.hs

```diff
@@ -77,8 +79,17 @@ options = Options
           <*> option (readP parse) (long "compiler" <> help "compiler to use when evaluating the Cabal file" <> value buildCompilerId <> showDefaultWith display)
           <*> option (readP parsePlatform) (long "system" <> help "target system to use when evaluating the Cabal file" <> value buildPlatform <> showDefaultWith display)
           <*> optional (strOption $ long "subpath" <> metavar "PATH" <> help "Path to Cabal file's directory relative to the URI (default is root directory)")
+          <*> optional (option utcTimeReader (long "hackage-snapshot" <> help "hackage snapshot time, example '09.08.2012 10:54 AM'"))
           <*> strArgument (metavar "URI")
 
+-- | A parser for the date. Minutes seem to be a good fit if the change is on Hackage is maybe once or twice a month.
+-- Example: parseTime defaultTimeLocale "%d.%m.%Y %l:%M %p" "09.08.2012 10:54 AM" :: Maybe UTCTime
```

Comment:
`cabal update` prints the appropriate datestamp in ISO format like this:

~~~
$ cabal update
Downloading the latest package list from hackage.haskell.org
To revert to previous state run:
    cabal update --index-state='2017-11-20T12:18:35Z'
~~~

Maybe we should accept (expect) that format, too?

Reply:
I'm wasn't sure about the format since seconds seem irrelevant, but for the sake of uniformity, I switched to an ISO format.